### PR TITLE
 Support set-command usage 

### PR DESCRIPTION
**Pull request by mckelvin**
ChangeLog:
- Build using cmake instead of autotools
- Fix IP header length
- Fix TCP header offset
- Support set-command usage (via `-C set`, for beansdb)
- Only filter packet sent from the captured network device for `get` command, and vice versa for `set` command.

cc @youngsofun @zzl0

### CODE REVIEW

**Comment by mckelvin** [comment]
@bmatheny I did not send the PR to the upstream since it's not active for a long time. Some commits in the PR are of personal favour (e.g.: use cmake instead of autotools). Please let me know if any of them should be sent to the upstream.

File: src/cli.cpp [hunk 1]
```diff
@@ -50,6 +51,10 @@ string Cli::help(const char * progname) {
           longopts[8],
           "Output data in REPORT format (CSV or curses, default curses)",
           "REPORT");
+  txt << mkHelpDoc(
+          longopts[9],
+          "Capture memcached command type (set or get, default get)",
+          "CAPTRUE_TYPE");
```
Comment (bmatheny) [inline src/cli.cpp]: Spelling.

Comment (mckelvin) [inline src/cli.cpp]: Thanks for your review! Please comment `lgtm` if the PR looks good to you.

``` diff
diff --git a/README.md b/README.md
index 798c385..be9a603 100644
--- a/README.md
+++ b/README.md
@@ -18,7 +18,7 @@ Build Status: [![Build Status](https://travis-ci.org/bmatheny/memkeys.png?branch
         -r, --refresh=INTERVAL      Refresh the stats display every INTERVAL ms (default 500)
         -l, --logfile=FILE          Output logs to FILE
         -R, --report=REPORT         Output data in REPORT format (CSV or curses, default curses)
-        -C, --capture=CAPTRUE_TYPE  Capture memcached command type (set or get, default get)
+        -C, --capture=CAPTURE_TYPE  Capture memcached command type (set or get, default get)

         -h, --help                  This help
         -v, --verbose               Increase verbosity. May be used multiple times.
diff --git a/src/cli.cpp b/src/cli.cpp
index 4233d2f..f046707 100644
--- a/src/cli.cpp
+++ b/src/cli.cpp
@@ -54,7 +54,7 @@ string Cli::help(const char * progname) {
   txt << mkHelpDoc(
           longopts[9],
           "Capture memcached command type (set or get, default get)",
-          "CAPTRUE_TYPE");
+          "CAPTURE_TYPE");
   txt << endl;
   txt << mkHelpDoc(longopts[4], "This help", "");
   txt << mkHelpDoc(longopts[5],
```

rebased and pushed --force

**Comment by bmatheny** [comment]
Please submit the non-build features upstream. I also don't think the address selection logic you have introduced is correct. If you're going to include the "Use default IP", please make it more generic. Either allow the user to specify an address on the command line, or don't include the change in the upstream pull request.

FWIW, not submitting because there hasn't been activity isn't a great reason. There would be activity if people submitted pull requests :)

**Comment by mckelvin** [comment]
@bmatheny Partial of these commits are from https://github.com/bmatheny/memkeys/pull/21 . Please consider merge that first. I'll check f433ef9 soon.

**Comment by mckelvin** [comment]
@bmatheny 

> I also don't think the address selection logic you have introduced is correct.

Agree. I can't find anything documentation about the order of the return value of pcap_findalldevs.  cmd line parameter `--address` is added. Please review again.

**Comment by mckelvin** [comment]
I will send a corresponding PR to upstream soon.

**Pull request merged**